src/bin/pgcopydb/ld_stream.c

```diff
@@ -1403,12 +1403,6 @@ streamFlush(LogicalStreamContext *context)
 {
 	StreamContext *privateContext = (StreamContext *) context->private;
 
-	/* when there is currently no file opened, just skip the flush operation */
-	if (privateContext->jsonFile == NULL)
-	{
-		return true;
-	}
-
```

Comment:
I think this only works because the later call to `streamKeepalive()` is in-turn going to call `streamRotateFile()` which ensures that a JSON file is now opened. Please add a comment either before calling `streamKeepalive()` or before calling `fileno` explaining this detail.